extension/content_script.js

```diff
@@ -149,18 +151,55 @@ function scroll() {
 Highlight portion pointed to by tracker.
 */
 function highlight(tracker) {
-	$(".marked").unmark();
-	$(".marked").removeClass("marked");
+
+	$("."+currentStyle).unmark();
+	$("."+currentStyle).removeClass(currentStyle);
 	// Append the "mark" class (?) to the html corresponding to the interval
 	// The interval indices are w.r.t to the raw text.
 	// mark.js is smart enough to preserve the original html, and even provide
 	// multiple consecutive spans to cover embedded htmls
-	tracker.getCurrentContainer().markRanges([{
-    	start: tracker.getStart(),
-    	length: tracker.getEnd() - tracker.getStart()
+	let container = tracker.getCurrentContainer();
+	let start = tracker.getStart();
+	let end = tracker.getEnd();
+	container.markRanges([{
+    	start: start,
+    	length: end - start
 	}], {
-		className: 'marked'
+		className: currentStyle
 	});
+	highlightKeyWords(container, start, end);
+};
+
+	/*
+    Params: Current container, start and end of tracker (jQuery element, int, int)
+    Highlights the keywords within the tracked sentence. 
+    */
+function highlightKeyWords(container, start, end) {
+	// TODO: Mark.js is actually built to do this; migrate functionality to mark.js
+	let keywordStyle = "keyWord";
+	$("."+keywordStyle).unmark(); // Remove previous sentence keyword styling
+	$("."+keywordStyle).removeClass(keywordStyle);
+	// Get list of words in interval
+	let containerText = container.text();
+	let sentenceText = containerText.slice(start,end);
+	let wordRegex = /\b\w+\b/g;
+	let wordList = sentenceText.match(wordRegex);
+	let keywords = doc.getKeyWords();
+	for (var i in wordList) { // Accentuate keywords
+		// TODO: This only gets the first occurence of each word in the sentence; should get all
+		let word = wordList[i];
+		if (keywords.includes(word.toLowerCase())) { // See if each word is a keyword
```

Comment:
prefer to use set() instead of array when you only want to check if the data structure contains an element
- https://alligator.io/js/sets-introduction/

array's hasElement() is O(n)
hashset's hasElement() is O(1) on average
--- If the above syntax isn't familiar, lmk and i can find useful resources.

Reply:
Oh yep, I thought about it for a second but was just trying to rapidly get to testing, definitely good to implement before master merge though

Also thanks for the js set link, I figured js had sets but was not directly familiar with

Reply:
Hmm I'm going to go to bed right now, can I address this in a separate pull request in the morning so that you can merge yours in as well (if you're going to tonight; if you're going to tomorrow no worries)?

I'll repush everything but this, and if you want you can just merge this and I'll do another CR tmrw mrnng
OR if you don't then I'll just address this then repush, CR again